companion/src/wizarddata.cpp

```diff
@@ -109,20 +108,26 @@ WizMix::operator ModelData()
   for (int i=0; i<WIZ_MAX_CHANNELS; i++ )
   {
     Channel ch = channel[i];
-    if (ch.input1 == THROTTLE_INPUT || ch.input2 == THROTTLE_INPUT)
-      throttleChannel = i;
 
     addMix(model, ch.input1, ch.weight1, i, mixIndex);
     addMix(model, ch.input2, ch.weight2, i, mixIndex);
-  }
 
-  // Add the Throttle Cut option
-  if( options[THROTTLE_CUT_OPTION] && throttleChannel >=0 ){
-    model.customFn[switchIndex].swtch.type = SWITCH_TYPE_SWITCH;
-    model.customFn[switchIndex].swtch.index = IS_ARM(getCurrentBoard()) ? SWITCH_SF0 : SWITCH_THR;
-    model.customFn[switchIndex].enabled = 1;
-    model.customFn[switchIndex].func = (AssignFunc)throttleChannel;
-    model.customFn[switchIndex].param = -100;
+    if (ch.input1 == THROTTLE_INPUT || ch.input2 == THROTTLE_INPUT) {
+      throttleChannel = i;
+
+      // Add the Throttle Cut option
+      if( options[THROTTLE_CUT_OPTION] && throttleChannel >=0 ){
```

Comment:
throttleChannel is always >= 0, no?

Reply:
It is IF you said your plane does have a motor. If you mistake and say  no motor but activate throttle cut, it would fail

Reply:
Hmm shouldn't throttle cut be unsllectable instead in this case then?